src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -4092,6 +4093,362 @@ export interface Format {
+  charismaLevel: number;
+  /** The depth into the darknet of the server. */
+  depth: number;
+  /** The model ID of the server. Similar models have similar vulnerabilities */
+  modelId: string;
+};
+
+/**
+ * Details about a server's authentication schema
+ * @public
+ */
+export type ServerAuthDetails = {
+  /** True if the server is still online. */
+  isOnline: boolean;
+  /** True if the server is directly connected to the current server */
+  isConnectedToCurrentServer: boolean;
+  /** True if the current script has authenticated to this server with the right password using authenticate() or connectToSesssion() */
+  hasSession: boolean;
+  /** The model ID of the server. Similar models share vulnerabilities. */
+  modelId: string;
+  /** Static password reminder text set for this server. */
+  passwordHint: string;
+  /** Data from the passwordHint, if any. */
+  data: string;
+  /** The frequency (in seconds) of the server adding its own messages to its logs, visible with heartBleed(). */
+  logTrafficInterval: number;
+  /** The number of characters in the password */
+  passwordLength: number;
+  /** The character set used in the password */
+  passwordFormat: "numeric" | "alphabetic" | "alphanumeric" | "ASCII" | "unicode";
+};
+
+/**
+ * Response from attempting to scrape logs from a server
+ * @public
+ */
+export type HeartbleedOptions = {
+  /**
+   * If true, looks at the most recent log line but does not extract it. (Overrides logsToCapture.)
+   */
+  peek?: boolean;
+  /**
+   * The number of log lines to remove from the server, up to a max of 8. Default is 1.
+   */
+  logsToCapture?: number;
+  /**
+   * The number of additional milliseconds to add to the run time of the heartbleed request. Default is 0.
+   */
+  additionalMsec?: number;
+};
+
+/**
+ * Darknet API
+ * @remarks
+ * If you are not in BitNode-15, then you must have Source-File 15 in order to use this API.
+ * @public
+ */
+export interface Darknet {
+  /**
+   * Sends a network request to try to authenticate on a darkweb server. The target server must be directly connected
+   * to the server that the script is running on. The speed of authentication scales with the number of threads used.
+   *
+   * If successful, grants the script a session, allowing it to exec() scripts on that server, or scp() files from it.
+   *
+   * Response messages:
+   * "200 Success" - Authentication was successful.
+   * "401 Unauthorized" - Authentication failed. The password is incorrect.
+   * "404 Not Found" - The server was not found. The server may be offline or the hostname is invalid.
+   * "408 Request Timeout" - The request failed (though the password may or may not have been correct). Caused by network instability.
+   * "301 Moved Permanently" - The server has moved to a different location and is no longer connected to the current server.
+   * "418 I'm a teapot" - The server is a teapot and cannot brew coffee.
```

Comment:
Here's one possibility
```suggestion
   * - "200 Success" - Authentication was successful.
   * - "401 Unauthorized" - Authentication failed. The password is incorrect.
   * - "404 Not Found" - The server was not found. The server may be offline or the hostname is invalid.
   * - "408 Request Timeout" - The request failed (though the password may or may not have been correct). Caused by network instability.
   * - "301 Moved Permanently" - The server has moved to a different location and is no longer connected to the current server.
   * - "418 I'm a teapot" - The server is a teapot and cannot brew coffee.
```